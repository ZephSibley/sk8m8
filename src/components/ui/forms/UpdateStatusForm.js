import React, { useState, } from 'react';
import { Formik } from 'formik';
import *  as Yup from 'yup';
import TextField from '@material-ui/core/TextField';
import FormHelperText from '@material-ui/core/FormHelperText';
import Button from '@material-ui/core/Button';

import formStyles from '../../../styles/forms';
import BackendValidationError from '../text/BackendValidationError';


const UpdateStatusForm = props => {
    // Prop: requests; http client
    // Prop: currentStatus; string

    const [submitError, setSubmitError] = useState([])

    const formClasses = formStyles();
    return (
        <div>
            <Formik
                initialValues={{ status: props.currentStatus || 'Whatcha sayin?', }}
                onSubmit={(values, { setSubmitting }) => {
                    console.log(values)
                    setSubmitting(true);
                    props.requests.post(
                        `${process.env.REACT_APP_ENDPOINT}/account/status`,
                        values
                    ).then(response =>
                        setSubmitting(false)
                    ).catch(err => {
                        setSubmitting(false);
                        setSubmitError(Object.values(JSON.parse(err)));
                    })
                }}

                render={({
                    values,
                    errors,
                    touched,
                    handleChange,
                    handleBlur,
                    handleSubmit,
                    isSubmitting,
                    isValid,
                }) => (
                        <form
                            onSubmit={handleSubmit}
                            className={formClasses.textFieldForm}
                        >
                            <TextField
                                name='status'
                                label='status'
                                onChange={handleChange}
                                onBlur={handleBlur}
                                value={values.status}
                                margin='normal'
                                multiline
                                rows="4"
                                error={errors.status && touched.status}
                                helperText={
                                    errors.status && touched.status ?
                                        errors.status : ''
                                }
                            />
                            <BackendValidationError resp={submitError} />
                            <Button
                                variant="outlined"
                                color="primary"
                                component="button"
                                className={formClasses.pullRight}
                                type="submit"
                                margin='normal'
                                disabled={isSubmitting}
                            >
                                Update
                            </Button>
                        </form>
                    )}
            />
        </div>
    )
}

export default UpdateStatusForm;