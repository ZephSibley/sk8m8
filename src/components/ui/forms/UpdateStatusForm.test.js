import React from 'react';
import renderer from 'react-test-renderer';
import mockAxios from '../../../utils/__mocks__/mockAxios';
import UpdateStatusForm from './UpdateStatusForm';

test('Basic Rendering', () => {
    //  TODO: 6Change to Post
    mockAxios.get.mockImplementationOnce(() =>
        Promise.resolve({
            data: {
                name: 'Name!',
                locationType: 'Skatey',
                creator: 'Zeph!'
            }
        })
    )

    const component = renderer.create(
        <UpdateStatusForm
            requests={mockAxios}
            currentStatus='Hey hey hey'
        />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})