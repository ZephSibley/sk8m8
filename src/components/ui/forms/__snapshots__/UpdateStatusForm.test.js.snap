// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div>
  <form
    className="makeStyles-textFieldForm-3"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Hey hey hey
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          disabled={false}
          name="status"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          rows="4"
          value=""
        />
      </div>
      
    </div>
    <pre
      className="MuiFormHelperText-root Mui-error"
    />
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-pullRight-7 MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      margin="normal"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Update
      </span>
    </button>
  </form>
</div>
`;
